forc-pkg/src/manifest.rs

```diff
@@ -127,6 +128,32 @@ impl ManifestFile {
     }
 }
 
+impl TryInto<PackageManifestFile> for ManifestFile {
+    type Error = anyhow::Error;
+
+    fn try_into(self) -> Result<PackageManifestFile> {
+        match self {
+            ManifestFile::Package(pkg_manifest_file) => Ok(*pkg_manifest_file),
+            ManifestFile::Workspace(_) => {
+                bail!("Cannot convert workspace manifest to package manifest")
+            }
+        }
+    }
+}
+
+impl TryInto<WorkspaceManifestFile> for ManifestFile {
+    type Error = anyhow::Error;
+
+    fn try_into(self) -> Result<WorkspaceManifestFile> {
+        match self {
+            ManifestFile::Package(_) => {
+                bail!("Cannot convert package manifest to workspace manifest")
+            }
+            ManifestFile::Workspace(workspace_manifest_file) => Ok(workspace_manifest_file),
+        }
+    }
+}
+
```

Comment:
I am a little unsure if we should have `TryInto` impls for this type of operations (converting an enum into its variants). In the compiler codebase we have `as_workspace()` etc for these type of conversions. 

Typically a `TryInto` is for converting a type into another one so it might be a little out of convention to use it for converting an enum to one of its variants.

Reply:
Does this look better? https://github.com/FuelLabs/sway/pull/5625